src/JsonApi/Serializer/CollectionNormalizer.php

```diff
@@ -82,9 +82,12 @@ protected function getItemsData($object, string $format = null, array $context =
 
         foreach ($object as $obj) {
             $item = $this->normalizer->normalize($obj, $format, $context);
+            if (!\is_array($item)) {
+                throw new UnexpectedValueException('Expected item to be an array');
+            }
 
             if (!isset($item['data'])) {
-                throw new InvalidArgumentException('The JSON API document must contain a "data" key.');
+                throw new UnexpectedValueException('The JSON API document must contain a "data" key.');
```

Comment:
Could it be a BC break?

Reply:
I'd consider it a bug fix, because we're not supposed to throw this type of exception from `NormalizerInterface::normalize` in the first place.